tools/twix/src/configuration.rs

```diff
@@ -0,0 +1,113 @@
+use std::path::{Path, PathBuf};
+
+use serde::Deserialize;
+use thiserror::Error;
+
+pub mod keybind_plugin;
+pub mod keys;
+
+const DEFAULT_CONFIG: &str = include_str!("../default.toml");
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+    #[error("TOML parsing error: {0}")]
+    Parsing(#[from] toml::de::Error),
+}
+
+fn config_path() -> PathBuf {
+    let mut result = dirs::config_dir().unwrap();
+    result.extend(["twix", "config.toml"]);
+
+    result
+}
+
+#[cfg_attr(test, derive(PartialEq))]
+#[derive(Debug, Deserialize)]
+pub struct Configuration {
+    pub keys: keys::Keybinds,
+}
+
+impl Configuration {
+    pub fn load() -> Result<Self, Error> {
+        Self::load_at_path(config_path())
+    }
+
+    pub fn load_at_path(path: impl AsRef<Path>) -> Result<Self, Error> {
+        match std::fs::read_to_string(&path) {
+            Ok(config_file) => {
+                let mut configuration = Self::load_default();
+                let user_configuration: Configuration = toml::from_str(&config_file)?;
+
+                configuration.merge(user_configuration);
```

Comment:
I tried matching helix' behavior here. This way, user configs can be relatively small if only a few keys are remapped, and you don't have to copy the entire default config (and keep it up-to-date with new features!) in order to remap single keys. I'm open for a debate here, it might be worth asking other team members what they think about this.

Reply:
Alright, @schmidma agrees with the current approach.
Gekauft.